doc/architecture/adr-002.md

```diff
@@ -0,0 +1,69 @@
+
+2. The current implementation is likely non-final: while the payment schedule specification is likely final, 
+some peripheral business logic (e.g. [the collection retry mechanism](https://github.com/crypto-org-chain/chain-main/issues/478))
+may need changes. If the implementation is put up on the live network, each implementation change will be 
+a consensus-breaking change and require synchronization via the gov and upgrade modules.
+
+3. The code requires non-trivial Gas metering explicit specifications: as the current implementation is likely non-final, one will also need to be careful to adjust the Gas meter logic correctly with each change.
+
+Overall, subscriptions are useful, but their current implementation may need to be re-assessed.
+
+## Decision
+For the reasons outlined in the context, the subscription module is to be removed from the current repository and its related issues can be closed.
+When the CosmWasm module is added (see [ADR-001](./adr-001.md)), the subscription module logic can be re-implemented in Rust
+as a CosmWasm smart contract that will be a part of a separate repository dedicated to advanced 
+smart contract implementations (crypto-org-chain/cosmwasm-plus-plus).
+Before the smart contract is deployed on the live network, its upgrade approach may need to consider the following options for the initial version:
+
+1. Restricting addresses that can instantiate the contract (`instantiate-only-address`), so that the reach is limited to early adopters before the implementation gets more finalized.
+
+2. Setting the admin address in the contract info, so that it can later be upgraded using the `MsgMigrateContract` message.
+
+3. Clearing the admin address and upgrading the contract via the `MigrateContractProposal` governance proposal.
+
+In the future, when the smart contract implementation is stabilized, it may be considered to bring back the revised native module
+implementation (if e.g. CosmWasm never intends to support triggering contracts by events rather than transactions and the on-chain automated collection is strongly desired).
+
+## Status
+
+Proposed
+
+## Consequences
+
+### Positive
+* Sandboxed environment
+* Rust compiler and Rust-related verification tools could be utilized for higher assurance
+* Implicit gas metering
+* More light-weight and flexible approach to upgrades
+* CW20 tokens could be used
+
+### Negative
+* Only manual payment collection: merchants or payment providers will need to set up their own cronjobs to submit transactions periodically
+* Complexity on the wallet implementation: native tokens need to be explicitly moved to the subscription contract in advance
+* Execution speed is potentially slower than on the native module
+* Potentially higher storage requirements
+* Complex audit trail: if one wants to fully verify the functionality, one will need to inspect Cosmos SDK, CosmWasm module, Wasmer interpreter, Rust compiler and Wasm optimizer, and CosmWasm smart contract
```

Comment:
In this case, who will be the auditor? And how much confidence we can trust the audit?

Reply:
This is meant in general -- but yes, if one wants to hire third party auditors, it may be hard to find one given these expertise requirements